cartography/intel/crxcavator/crxcavator.py

```diff
@@ -49,55 +52,142 @@ def call_crxcavator_api(api_and_parameters, crxcavator_api_key, crxcavator_base_
     """
     Transforms the raw extensions JSON from the API into a list of extensions data
-    :param extension_json:  The JSON text blob returned from the CRXcavator API
+    :param extension_details:  List containing the extension details
     :return: List containing extension info for ingestion
     """
     # the JSON returned from the CRXcavator API does not return well formatted objects
     # instead, each object is named after it's key, making enumeration more difficult
     # will build a cleaner object for import into graph
 
-    _data_index = 1
-
     extensions = []
-    for extension in extension_json.items():
-        for details in extension[_data_index]:
-            if not details:
-                logger.warning(f'Could not retrieve details for extension {extension}')
-                continue
-            extension_id = details['extension_id']
-            version = details['version']
-            data = details['data']
-            extensions.append({
-                'id': f"{extension_id}|{version}",
-                'extension_id': extension_id,
-                'version': version,
-                'risk_total': data['risk'].get('total'),
-                'risk_metadata': json.dumps(data['risk'].get('metadata')),
-                'address': data['webstore'].get('address'),
-                'email': data['webstore'].get('email'),
-                'icon': data['webstore'].get('icon'),
-                'crxcavator_last_updated': data['webstore'].get('last_updated'),
-                'name': data['webstore'].get('name'),
-                'offered_by': data['webstore'].get('offered_by'),
-                'permissions_warnings': data['webstore'].get('permission_warnings'),
-                'privacy_policy': data['webstore'].get('privacy_policy'),
-                'rating': data['webstore'].get('rating'),
-                'rating_users': data['webstore'].get('rating_users'),
-                'short_description': data['webstore'].get('short_description'),
-                'size': data['webstore'].get('size'),
-                'support_site': data['webstore'].get('support_site'),
-                'users': data['webstore'].get('users'),
-                'website': data['webstore'].get('website'),
-                'type': data['webstore'].get('type'),
-                'price': data['webstore'].get('price'),
-                'report_link': "https://crxcavator.io/report/" + extension_id + "/" + version,
-            })
+    for extension in extension_details:
+        extension_id = extension['extension_id']
+        version = extension['version']
+        data = extension.get('data')
+        if not data:
+            logger.warning(f'Could not retrieve details for extension {extension}')
+            continue
+        risk_data = data.get('risk')
+        if not risk_data:
+            risk = {}
+        else:
+            risk = parse_risk(risk_data)
+        webstore = data.get('webstore')
```

Comment:
Since `webstore` is used below, this needs to be `data.get('webstore', {})`

Reply:
I checked for null immediately after this, but went ahead and changed anyways.